sky/serve/__init__.py

```diff
@@ -6,11 +6,6 @@
 from sky.serve.constants import LB_CONTROLLER_SYNC_INTERVAL_SECONDS
 from sky.serve.constants import SERVICES_TASK_CPU_DEMAND
 from sky.serve.constants import SKYSERVE_METADATA_DIR
-from sky.serve.core import down
-from sky.serve.core import status
-from sky.serve.core import tail_logs
-from sky.serve.core import up
-from sky.serve.core import update
```

Comment:
We temporarily removed these imports to avoid the circlic imports...

Reply:
cc'ing @cblmemo @MaoZiming. Could you help see how we can solve this circlic imports?

Reply:
Just took a look, and in the latest commit (`6a10a6f4`), it could be resolved by the following three steps:
1. Remove `from sky import optimizer` from `sky/backends/cloud_vm_ray_backend.py`
  * The only place used this import is as follows, and we could use `None` as the default value https://github.com/skypilot-org/skypilot/blob/70a7435581c35fb76dbe3c1317136ec141937d1e/sky/backends/cloud_vm_ray_backend.py#L2552
2. Remove `from sky.backends import backend_utils` from `sky/utils/cli_utils/status_utils.py`
  * The only place used is as folows and we could deprecate this function (since local cluster is deprecated?) or pass in the result of `backend_utils.get_clusters` in the caller of this function https://github.com/skypilot-org/skypilot/blob/70a7435581c35fb76dbe3c1317136ec141937d1e/sky/utils/cli_utils/status_utils.py#L203-L217
And the final step has two choices, either would work:
3. (1) Remove `from sky import spot` from `sky/resources.py`
  * The only place used this import is as follows, and we could have another constant list maintained in another file (as long as not in the `sky.spot.recovery_strategy` https://github.com/skypilot-org/skypilot/blob/70a7435581c35fb76dbe3c1317136ec141937d1e/sky/resources.py#L736-L749
3. (2) Remove `from sky import serve as serve_lib` and `from sky.utils import controller_utils` in `sky/backends/backend_utils.py`
  * The only usage of `serve_lib` is `SERVICES_TASK_CPU_DEMAND`, and we could move this constant to other files like `sky.skylet.constants`
  * The only usage of `controller_utils` is the `include_controller` argument in `get_clusters` function, and we could do this filtering in caller function https://github.com/skypilot-org/skypilot/blob/70a7435581c35fb76dbe3c1317136ec141937d1e/sky/backends/backend_utils.py#L2325-L2359

Reply:
In 3.(2) we should import the `controller_utils` inside the `if typing.TYPE_CHECKING:` instead for type notation in this func: https://github.com/skypilot-org/skypilot/blob/70a7435581c35fb76dbe3c1317136ec141937d1e/sky/backends/backend_utils.py#L2244-L2245

Reply:
Thanks for the detailed solution @cblmemo! I just tried another way, but moving the following line from the `sky.serve.constants` to `sky.skylet.constants`, and seems fixing the issue:
```python
SERVICES_MEMORY_USAGE_GB = 1.0
SERVICES_TASK_CPU_DEMAND = 0.25
```
We can try to refactor them back in the future, wdyt?

Reply:
sounds good!